### PR TITLE
Private event features

### PR DESCRIPTION
**Pull request by AdfPizarro**
You can sign in and login
-Create events
-View all the events
-See the past and upcoming events
-Attend to events
-See your appointments

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 class User < ApplicationRecord
-     has_many :events, foreign_key: :creator
-     has_many :attendees, foreign_key: :attendee_id
-#    has_many :appointments, class_name Event.name,  foreign_key: : 
-     has_many :appointments, through: :attendees, source: :attended_event
+  has_many :events, foreign_key: :creator
```
Comment (addod19) [inline app/models/user.rb:2]: Kindly replace the `:creator` with `:creator_id`

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -1,42 +1,41 @@
...
+        format.json { render json: @user.errors, status: :unprocessable_entity }
       end
     end
+  end
 
-    def show
-     @events = current_user.appointments 
-    end
+  def show
+    @events = current_user.appointments
```
Comment (addod19) [inline app/controllers/users_controller.rb:33]: `UserController` show; 
- list events created by user, ie. `@created_events = user.events`
- list events previously attended by user, using event scope `@past_events = user.attended_events.previous`
- list upcoming event previously attended by user, using event scope `@past_events = user.attended_events.previous`

**Review by addod19 — CHANGES_REQUESTED** [review]
## Status: Require changes :recycle: :traffic_light: 

Hi @AdfPizarro, you did a great job on this project :+1: Awesome!!! However, there a few comments we need to address to get our project approved :100: Kindly check the comments below:
- `Events show` page should display the list of attendants. There is currently no button for users to click to indicate whether they are attending or not. Let us add a button action for that :slightly_smiling_face: 
- `Users show` page should display events the user has created and events that the user is attending to separated in `past` and `upcoming`. Currently, this is what we have. Check image below
sh
- We are currently missing a test for our project. Let us implement test
tsss
- Kindly remove all commented code, code commented should be removed

[Optional]
- Kindly the association of the `attendee` model, it can be improved

Once this is resolved, kindly submit another code review :smile: 
_Happy coding_ :man_technologist: 
[Daniel Larbi Addo](https://www.linkedin.com/in/daniel-larbi-addo/)

File: app/controllers/users_controller.rb (hunk 2 above)
Comment (AdfPizarro) [reply app/controllers/users_controller.rb:36]: Added the display in the user show page

File: app/models/user.rb (hunk 1 above)
Comment (AdfPizarro) [reply app/models/user.rb:2]: Renamed and removed the commented line

**Review by sumancrest0001 — APPROVED** [review]
**Status: Approved:100: 
Congratulations @AdfPizarro :clap: :clap: **
Thank you for making all the changes requested by the previous TSE. It is approved. Now you can move to the next project.
Happy coding !!!

**Pull request merged**